src/main/java/com/group8/library_management/service/impl/UserServiceImpl.java

```diff
@@ -12,4 +22,53 @@ public UserServiceImpl(UserRepository userRepository) {
         this.userRepository = userRepository;
     }
 
+    @Override
+    public Page<User> getAllUsers(int page, int size, String searchName, String sortBy) {
+        Sort sort = Sort.unsorted();
+        if (sortBy != null && !sortBy.isEmpty()) {
+            if (sortBy.contains(",")) {
+                String[] parts = sortBy.split(",");
+                sort = Sort.by(Sort.Direction.fromString(parts[1].toUpperCase()), parts[0]);
+            } else {
+                sort = Sort.by(sortBy).ascending();
+            }
+        } else {
+            // Sắp xếp mặc định nếu không có tùy chọn sortBy được cung cấp
+            sort = Sort.by("id").ascending();
+        }
+        Pageable pageable = PageRequest.of(page, size, sort);
+
+        Specification<User> spec = (root, query, cb) -> {
+            if (searchName != null && !searchName.isEmpty()) {
+                return cb.like(cb.lower(root.get("fullName")), "%" + searchName.toLowerCase() + "%");
+            }
+            return null;
+        };
```

Comment:
```suggestion
        Specification<User> spec = (root, query, cb) -> {
        if (searchName != null && !searchName.isEmpty()) {
            return cb.like(cb.lower(root.get("fullName")), "%" + searchName.toLowerCase() + "%");
        }
        return cb.conjunction();
    };
```
1. Dùng trim() để loại khoảng trắng
2. Dùng method  conjunction() giúp tránh return khi không có điều kiện tìm kiếm

Reply:
cơ mà query mình nên call ở repository -> sugest thôi